dpctl/_sycl_queue_manager.pyx

```diff
@@ -223,25 +223,22 @@ def device_context(arg):
     usable queue on exiting the context manager.
 
     Args:
-
-        queue_str (str) : A string corresponding to the DPC++ filter selector.
+        arg : A :class:`.SyclDevice` object, or a :class:`.SyclQueue` object, or a filter selector string
 
     Yields:
-        :class:`.SyclQueue`: A SYCL queue corresponding to the specified
-        filter string.
+        :class:`.SyclQueue`: A SYCL queue corresponding to the specified input device, queue, or filter string.
 
     Raises:
         SyclQueueCreationError: If the SYCL queue creation failed.
 
     :Example:
-        To create a scope within which the Level Zero GPU number 0 is active,
-        a programmer needs to do the following.
+        The following example activates Level Zero GPU number 0 by respective filter selector string:
```

Comment:
```suggestion
        The following example sets current queue targeting specific device indicated with filter selector string in the scope of `with` block:
```

Reply:
Note that the whole concept of `device_context` and queue manager is de-facto deprecated and will become obsolete once compute follows data is embraced.

Reply:
@oleksandr-pavlyk, yes, the compute follows data will make device_context() almost obsolete. But what about scenario when I created NumPy array with NumPy on the host but want the kernel to be compiled for GPU with Numba, data inherently converted from NumPy array to underlying USM array?

import dpctl
import numpy as np
from numba import njit

@njit
def do_compute(a):
    return np.log(a+1.0)

a = np.arange(10)
with dpctl.device_context(dpctl.select_gpu_device())
    b = do_compute(a)
print(b)

I find it very convenient. I remember multiple discussions about the compute follow data but do not remember we agreed on deprecation of the device context mechanism.